src/charts/helpers/export.js

```diff
@@ -54,13 +56,21 @@ export function exportChart(d3svg, filename, title) {
         return false;
     }
 
-    let img = createImage(convertSvgToHtml.call(this, d3svg, title));
-
-    img.onload = handleImageLoad.bind(
-        img,
-        createCanvas(this.width(), this.height()),
-        filename
-    );
+    return loadImage(convertSvgToHtml.call(this, d3svg, title))
+        .then((img) => {
+            return {
+                canvas: createCanvas(this.width(), this.height()),
+                img,
+            };
+        })
+        .then(({ canvas, img }) => handleImageLoad.call(img, canvas, filename))
+        .then(() => true)
```

Comment:
Why do we need this? is for defaulting to success? Not sure if I understand it.

Reply:
Great question. TBH we do not need this. I will remove it and let the user decide what the user wants to do. What do you think?

Reply:
maybe is the same question that @Golodhros asked, but a `Promise.resolve` or a `Promise.reject` should be the replacement for a true-false behavior.

Reply:
@gagoar great point. I believe the Promise constructor is already handling the `Promise.resolve` and `Promise.reject`. So if I remove that last `true/false` thing then the user will already bee able to have a `.then` callback if it's successful and `.catch` if it's a `Promise.reject`. Thanks for feedback!